hw/drivers/ipc_nrf5340/syscfg.yml

```diff
@@ -48,6 +48,17 @@ syscfg.defs:
             eg "LED_1, LED_2" or "1, 2". Further GPIO configuration should be
             done by Network Core.
         value: ""
+    IPC_NRF5340_PRE_TRUSTZONE_NETCORE_BOOT:
+        description: >
+            Set this to one only in case of the bootloader predates TrustZone
+            changes. With ARM TrusZone support added NRF_IPC_NS block is used
+            for both secure and non-secure application. Pre-TrustZone code
+            was always running in secure mode and network core code always
+            accessed NRF_IP_S (including code that is running in bootloader).
+            NRF_IP_S->GPMEM[] was used to pass address of net core application
+            image to net bootloader.
+        value: 0
 
 syscfg.restrictions:
     - "!BSP_NRF5340 || BSP_NRF5340_NET_ENABLE"
+    - "!IPC_NRF5340_PRE_TRUSTZONE_NETCORE_BOOT || MCU_APP_SECURE"
```

Comment:
Given that `MCU_APP_SECURE` is undefined on the netcore side, you would hit this restriction if you set `IPC_NRF5340_PRE_TRUSTZONE_NETCORE_BOOT` to 1